plugins/experimental/wasm/ats_context.cc

```diff
@@ -1325,6 +1331,10 @@ Context::getProperty(std::string_view path, std::string *result)
     int64_t total   = h_bytes + b_bytes;
     result->assign(reinterpret_cast<const char *>(&total), sizeof(int64_t));
     return WasmResult::Ok;
+  } else if (path.substr(0, p_txn_is_internal.size()) == p_txn_is_internal) {
+    int internal = TSHttpTxnIsInternal(txnp_);
+    result->assign(reinterpret_cast<const char *>(&internal), sizeof(int));
+    return WasmResult::Ok;
```

Comment:
I am actually thinking of using foreign function instead for this. e.g. in envoy it has "compress and "uncompress" foreign function exposed to wasm modules.

This is how it is defined - 
https://github.com/envoyproxy/envoy/blob/d12d47b05a832ce04d9130a869fa72f7ddbcef28/source/extensions/common/wasm/foreign.cc#L43

This is how it is used -
https://github.com/envoyproxy/envoy/blob/d12d47b05a832ce04d9130a869fa72f7ddbcef28/test/extensions/common/wasm/test_data/test_cpp.cc#L158

Please take a look and see if we can implement this using foreign function